src/client/game.cpp

```diff
@@ -1238,11 +1237,16 @@ void Game::shutdown()
 	if (g_touchscreengui)
 		g_touchscreengui->hide();
 
-	showOverlayMessage(N_("Shutting down..."), 0, 0, false);
-
 	if (clouds)
 		clouds->drop();
 
+	FpsControl fps_control;
+	fps_control.reset();
+	f32 dtime;
+	m_rendering_engine->run();
+	fps_control.limit(device, &dtime);
```

Comment:
fps_control and `m_rendering_engine->run` have no point unless they're being run in a loop.
since you're just drawing a single frame here you can assume `dtime=0`

Reply:
could also simply keep the code as it was before

Reply:
@grorp like I said previously already, that leads to a flashing screen

Reply:
You mean this, right?

> However, having this results in a black screen that is only displayed for the fraction of a second on my computer…

The fix is probably to pass true to the draw_clouds argument of setOverlayMessage.